### ISSUE TITLE
Support virtual hosts

### ISSUE THREAD

**Issue by danlangford** [noodlefrenzy/node-amqp10#171]
I have already talked to the devs a bit in gitter.im about supporting Virtual Hosts but i wanted to create a longer lasting record of the work being done around this. Especially since i may send other people to this ticket for information. 

We would like to use this client lib to connect to our Qpid Java Broker. Qpid Java brokers must be setup with a virtual host. Out of the box the main virtual host is called "default"

Initially i couldnt see how to specify a remote host in the API. some of the URI patterns i have seen in other libs look like this:
 - `amqps://user:pass@example.com/default`
 - `amqps://user:pass@example.com?amqp.vhost=default`
 - `amqps://user:pass@clientid/default?brokerlist='tcp://example.com:5671'`

in our discussions @mbroadst ended up creating a branch `java-vhosts` which includes some attempts to support virtual hosts using the first pattern mentioned above. as of this writing it actually works but he indicated he wanted to clean it up a little and finalize the solution differently. i hope that he will comment with the details of what he needed to change but it had something to do with the `hostname` field in the init frames.... i think

**Comment by danlangford** [comment]
FYI @mbroadst i have been also working on this with Azure/amqpnetlite#36 and he showed me how the lib can currently specify a vhost. what was interesting is that it works fine with only the hostname field set in the open performative not in the sasl init frame. here is some output from my c# tests:

```
SEND AMQP 3 1 0 0
SEND sasl-init(mechanism:PLAIN,initial-response:<redacted>,hostname:example.com)
RECV AMQP 3 1 0 0
RECV sasl-mechanisms(sasl-server-mechanisms:[PLAIN])
RECV sasl-outcome(code:0)
SEND AMQP 0 1.0.0
SEND (ch=0) open(host-name:default)
SEND (ch=0) begin(next-outgoing-id:4294967293,incoming-window:2048,outgoing-window:2048,handle-max:63)
SEND (ch=0) attach(name:send-link,handle:0,role:False,source:source(),target:target(address:my-queue),initial-delivery-count:0)
```

thought that you may like to know in case that helps clarify or simplify anything. i do not know what is the "right" way and we have shown that it doesnt break anything to have that in the SASL init frame. 

so ya, fyi. thanks

**Comment by mbroadst** [comment]
fixed with 119bf49245d67fb90444308a1c314dea99a6f7ff
